### PR TITLE
added a migration for `InferenceByEpisodeId` table

### PR DESCRIPTION
**Pull request by virajmehta**
> [!IMPORTANT]
> Add `Migration0007` for `InferenceByEpisode` table with materialized views and update rollback instructions in `migration_0001.rs`.
> 
>   - **Migration**:
>     - Add `Migration0007` in `migration_0007.rs` to create `InferenceByEpisode` table and materialized views for efficient querying by episode ID.
>     - Update rollback instructions in `migration_0001.rs` to drop views instead of materialized views.
>   - **Integration**:
>     - Include `Migration0007` in `mod.rs` and `clickhouse.rs` for execution and testing.
>   - **Testing**:
>     - Add tests for `Migration0007` in `clickhouse.rs` to ensure proper application and rollback.
> 
> <sup>This description was created by </sup>[Ellipsis](https://www.ellipsis.dev?ref=tensorzero%2Ftensorzero&utm_source=github&utm_medium=referral)<sup> for 2b7dedbc0373aabe00f59c61dcb06d6c8d27a969. It will automatically update as commits are pushed.</sup>

### CODE REVIEW

File: gateway/src/clickhouse_migration_manager/migrations/migration_0007.rs [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+
+        // Create the `InferenceByEpisode` table
+        let query = r#"
+            CREATE TABLE IF NOT EXISTS InferenceByEpisode
+            (
+                episode_id UUID,
+                id UUID,
+                function_name LowCardinality(String),
+                variant_name LowCardinality(String),
+                function_type Enum8('chat' = 1, 'json' = 2),
```
[the remark this answers is not in the document]
Comment (virajmehta) [reply gateway/src/clickhouse_migration_manager/migrations/migration_0007.rs:86]: passes `rustfmt` so I assume this is wrong

File: gateway/src/clickhouse_migration_manager/migrations/migration_0007.rs [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+                    id: "0007".to_string(),
+                    message: e.to_string(),
+                })
+            })?
+            + view_offset)
+            .as_secs();
+
+        // Create the `InferenceByEpisode` table
+        let query = r#"
+            CREATE TABLE IF NOT EXISTS InferenceByEpisode
```
Comment (GabrielBianconi) [inline gateway/src/clickhouse_migration_manager/migrations/migration_0007.rs:80]: `InferenceByEpisodeId` & similar changes throughout

**Review by GabrielBianconi — CHANGES_REQUESTED** [review]
Minor naming change for clarity.

File: gateway/src/clickhouse_migration_manager/migrations/migration_0007.rs (hunk 2 above)
Comment (virajmehta) [reply gateway/src/clickhouse_migration_manager/migrations/migration_0007.rs:80]: kk will do

**Review by GabrielBianconi — APPROVED** [review]

**Pull request merged**